Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx

```diff
@@ -92,97 +112,270 @@ function sortDialog(dialogs: DialogInfo[]) {
   });
 }
 
-function createItemsAndGroups(
-  dialogs: DialogInfo[],
-  dialogId: string,
-  filter: string
-): { items: any[]; groups: IGroup[] } {
-  let position = 0;
-  const result = dialogs
-    .filter((dialog) => {
-      return dialog.displayName.toLowerCase().includes(filter.toLowerCase());
-    })
-    .reduce(
-      (result: { items: any[]; groups: IGroup[] }, dialog) => {
-        const warningContent = containUnsupportedTriggers(dialog);
-        result.groups.push(createGroupItem(dialog, dialogId, position, warningContent));
-        position += dialog.triggers.length;
-        dialog.triggers.forEach((item, index) => {
-          const warningContent = triggerNotSupported(dialog, item);
-          result.items.push(createItem(item, index, warningContent));
-        });
-        return result;
-      },
-      { items: [], groups: [] }
-    );
-  return result;
-}
-
-interface IProjectTreeProps {
+type BotInProject = {
   dialogs: DialogInfo[];
-  dialogId: string;
-  selected: string;
-  onSelect: (id: string, selected?: string) => void;
+  projectId: string;
+  name: string;
+  isRemote: boolean;
+};
+
+type IProjectTreeProps = {
+  onSelect?: (link: TreeLink) => void;
+  onAllSelected?: () => void;
```

Comment:
nit: onSelectAll (to follow your other callbacks naming pattern)

Reply:
I almost named it that, but that reads to me like there's some kind of "select-all" action  - as in "select everything in the tree somehow" when what I really mean is that the literal link named "All" got selected.

Reply:
I think I'll call this `onSelectAllLink`. That's different enough from "select all" that I think it reads better as "select the 'all' link".